Update --no-warn-no-return docs for empty body changes (#14065)
python/mypy: docs/source/command_line.rst

    are when:

    -   The function has a ``None`` or ``Any`` return type

>     -   The function has an empty body and is marked as an abstractmethod,

Reviewer: ```suggestion
    -   The function has an empty body and is marked as an abstract method,
```

        is in a protocol class, or is in a stub file

>     -  The execution path is unreachable; for example, if an exception

Author: ```suggestion
    -  The execution path can never return; for example, if an exception
```

>     -   For unreachable execution paths, for example, if an exception
>         is always raised

Reviewer: ```suggestion
    -  The execution path is unreachable; for example, if an exception
        is always raised
```
The list is introduced with "when", so each list item should complete the "when".
